### PR TITLE
Added a title message on join. Closes #679.

### LINKED ISSUE #679: Title on Join
On join, a message could pop up saying "Hello," etc.

### PR DESCRIPTION
**Pull request by Hockeyfan360**
This commit adds a title message on join. I fixed the issues with the other PR and fixed them.

### CODE REVIEW

**Comment by JeromSar** [comment]
@Hockeyfan360 As a rule of thumb, don't make new PRs for every fixup commit on an existing PR. Simply append your changes to the new PR.

File: src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java [hunk 1]
```diff
@@ -910,6 +910,8 @@ public void run()
                 }
             }
         }.runTaskLater(TotalFreedomMod.plugin, 20L * 1L);
+        TFM_ServerInterface.sendTitle(player, "§6§lWelcome back,", 2, 2, 2);
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java]: Please use `ChatColor` for colors.

File: src/me/StevenLawson/TotalFreedomMod/TFM_ServerInterface.java [hunk 2]
```diff
@@ -241,4 +247,41 @@ public static void handlePlayerLogin(PlayerLoginEvent event)
...
+            CraftPlayer craftplayer = (CraftPlayer) player;
+            PlayerConnection connection = craftplayer.getHandle().playerConnection;
+            String finalmessage = message.replaceAll("&", "§");
+            IChatBaseComponent chatTitle = ChatSerializer.a("{\"text\": \"" + finalmessage + "\"}");
+            PacketPlayOutTitle title = new PacketPlayOutTitle(EnumTitleAction.TITLE, chatTitle, fadein, stay, fadeout);
+            connection.sendPacket(title);
+        }
+        catch (Exception e)
+        {
+            TFM_Log.severe(e.getLocalizedMessage());
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/TFM_ServerInterface.java]: Please log the entire stacktrace here.

File: src/me/StevenLawson/TotalFreedomMod/TFM_ServerInterface.java [hunk 3]
```diff
@@ -241,4 +247,41 @@ public static void handlePlayerLogin(PlayerLoginEvent event)
...
+            TFM_Log.severe("Title packet send failed. Your server jar is likely outdated.");
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/TFM_ServerInterface.java]: For standardisation purposes, the "info" message should be sent before the stack trace. And it should describe only the problem that occurred, not the possible cause.

In this case, something along the lines of `"Could not send title packet to player " + player.getName()` should suffice.

**Comment by OutOfYourMind** [comment]
Is this needed?

**Comment by Hockeyfan360** [comment]
@OutOfYourMind The issue was accepted, so I went ahead and made it.

File: src/me/StevenLawson/TotalFreedomMod/TFM_ServerInterface.java [hunk 4]
```diff
@@ -241,4 +247,41 @@ public static void handlePlayerLogin(PlayerLoginEvent event)
...
     }
+    
+     //Please note that with titles, you must ALWAYS send the title first, and the subtitle second.
+    public static void sendTitle(Player player, String message, int fadein, int stay, int fadeout)
+    {
+        try
+        {
+            CraftPlayer craftplayer = (CraftPlayer) player;
+            PlayerConnection connection = craftplayer.getHandle().playerConnection;
+            String finalmessage = message.replaceAll("&", "§");
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/TFM_ServerInterface.java]: Please use `ChatColor.translateAlternateColorCodes()` here.

File: src/me/StevenLawson/TotalFreedomMod/TFM_ServerInterface.java [hunk 5]
```diff
@@ -241,4 +247,41 @@ public static void handlePlayerLogin(PlayerLoginEvent event)
...
+        }
+    }
+
+    public static void sendSubtitle(Player player, String message, int fadein, int stay, int fadeout)
+    {
+        try
+        {
+            CraftPlayer craftplayer = (CraftPlayer) player;
+            PlayerConnection connection = craftplayer.getHandle().playerConnection;
+            String finalmessage = message.replaceAll("&", "§");
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/TFM_ServerInterface.java]: Here too. :)

**Comment by OnlineRepz** [comment]
@Hockeyfan360, any updates?

**Comment by Hockeyfan360** [comment]
Please don't copy and paste my code and call it your work. Thx much.

**Comment by JeromSar** [comment]
@Hockeyfan360 I don't understand. Where did this happen?

**Comment by rsajeey** [comment]
@JeromSar @Hockeyfan360 I believe he thought that it was someone else's PR that had copied his code into the PR, but it was actually his own PR.

**Comment by Hockeyfan360** [comment]
@JeromSar

I've closed PRs before only to see that people copy and paste my code with the required edits. That's what I meant.

**Comment by JeromSar** [comment]
@Hockeyfan360 Ah right.

**Pull request closed without merging**